website_openstreetmap/models/res_company.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    This module copyright (C) 2015 Therp BV (<http://therp.nl>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api, exceptions, _
+import urllib
+import json
+
+
+class ResCompany(models.Model):
+
+    _inherit = "res.company"
+
+    gps_longitude = fields.Char('longitude gps',)
+    gps_latitude = fields.Char('latitude gps',)
+
+    def openstreetmap_img(self,  zoom=16, width=250, height=250):
+        url = 'http://staticmap.openstreetmap.de/staticmap.php'
+        params = {
+            'center': "{lat},{lon}".format(
+                lat=self.gps_latitude or '',
+                lon=self.gps_longitude or ''),
+            'size': "%sx%s" % (height, width),
+            'zoom': zoom,
+            'maptype': 'mapnik',
+            'map': "{zm}/{lat}/{lon}".format(
+                zm=zoom,
+                lat=self.gps_latitude or '',
+                lon=self.gps_longitude or ''),
+        }
+        return url + '?%s' % urllib.urlencode(params)
+
+    def openstreetmap_link(self, zoom=16):
+        url = 'http://www.openstreetmap.org'
+        params = {
+            'mlat': self.gps_latitude,
+            'mlon': self.gps_longitude,
+            'zoom': zoom,
+            'layers': 'M',
+            'map': "{zm}/{lat}/{lon}".format(
+                zm=zoom,
+                lat=self.gps_latitude or '',
+                lon=self.gps_longitude or '', )
+        }
+        return url + '?%s' % urllib.urlencode(params)
+
+    @api.multi
+    def get_nominatim_openstreetmap_coordinates(self):
```

Comment:
This kind of service is already provided by https://github.com/OCA/geospatial/blob/8.0/geoengine_base_geolocalize/models/res_partner.py. It would be nice to share the same code.